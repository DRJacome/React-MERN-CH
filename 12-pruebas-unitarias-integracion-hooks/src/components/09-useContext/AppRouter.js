import React from "react";
/* Para React Router v5.x -> usar Switch;
* Para React Router V6.x -> usar Routes.
 */
import {
  BrowserRouter as Router,
  Switch, //Routes,
  Route
} from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { LoginScreen } from "./LoginScreen";
import { HomeScreen } from "./HomeScreen";
import { NavBar } from "./NavBar";
import { Redirect } from "react-router-dom";


export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen} />
            <Route exact path="/about" component={AboutScreen} />
            <Route exact path="/login" component={LoginScreen} />
            <Redirect to='/' /> {/* <Route component={HomeScreen} /> */}
          </Switch>
        </div>
      </div>
    </Router>
  )
}