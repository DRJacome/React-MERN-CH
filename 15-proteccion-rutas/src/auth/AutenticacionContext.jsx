import React, { createContext } from 'react'

export const AutenticacionContext = createContext()