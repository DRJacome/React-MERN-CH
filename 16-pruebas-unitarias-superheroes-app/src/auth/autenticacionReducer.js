import { types } from '../types/types';

/* Estado inicial:
  const state = {
  nombre: 'Fernando',
  logged: true
} */

/* El par치metro state est치 definido fuera del reducer y el par치metro action est치 desestructurado 
para acceder directamente al type desde el switch */
export const autenticacionReducer = (state = {}, action) => {
  switch (action.type) {
    case types.login:
      return {
        ...action.payload,
        logged: true
      }
    case types.logout:
      return {
        logged: false
      }
    default:
      return state
  }
}
