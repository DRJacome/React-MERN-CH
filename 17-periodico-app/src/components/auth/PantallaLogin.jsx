import React from 'react'
import { Link } from "react-router-dom"

export const PantallaLogin = () => {
    return (
        <>
            <h3 className="auth__title">Login</h3>
            <form>
                <input
                    type="text"
                    placeholder="email"
                    name="email"
                    className="auth__input"
                    autoComplete="off"
                />
                <input
                    type="password"
                    placeholder="Contrase침a"
                    name="password"
                    className="auth__input"
                />
                <button
                    type="submit"
                    className="btn btn-primary btn-block"
                    disabled={true}
                >
                    Iniciar sesi칩n
                </button>
                <hr />
                <div className="auth__social-networks">
                    <p>Iniciar sesi칩n con RRSS</p>
                    <div
                        className="google-btn"
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="/assets/icono-google.png" alt="bot칩n google" />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with Google</b>
                        </p>
                    </div>
                </div>
                <Link to='/auth/register' className="link">Crear cuenta</Link>
            </form>
        </>
    )
}