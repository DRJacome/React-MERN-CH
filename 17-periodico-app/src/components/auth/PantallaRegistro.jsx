import React from 'react'
import { Link } from "react-router-dom"


export const PantallaRegistro = () => {
  return (
    <>
      <h3 className="auth__title">Registro</h3>
      <form>
        <input
          type="text"
          placeholder="Nombre"
          name="nombre"
          className="auth__input"
          autoComplete="off"
        />
        <input
          type="password"
          placeholder="Contraseña"
          name="password"
          className="auth__input"
        />
        <input
          type="password"
          placeholder="Confirmar contraseña"
          name="confirmarPassword"
          className="auth__input"
        />
        <button
          type="submit"
          className="btn btn-primary btn-block mb-5"
          disabled={true}
        >
          Registrarse
        </button>
        <hr />
        
          <Link to='/auth/login' className="link">¿Ya estás registrado?</Link>
      </form>
    </>
  )
}