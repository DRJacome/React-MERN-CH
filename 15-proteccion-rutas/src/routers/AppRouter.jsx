import React, { useContext } from "react";
import {
  BrowserRouter as Router,
  Switch
} from "react-router-dom";
import { VistaLogin } from "../components/login/VistaLogin";
import { PanelRutas } from "./PanelRutas";
import { RutaPrivada } from "./RutaPrivada";
import { RutaPublica } from "./RutaPublica";
import { AutenticacionContext } from "../auth/AutenticacionContext";

export const AppRouter = () => {
  const { usuario } = useContext(AutenticacionContext)
  console.log(usuario);

  return (
    <Router>
      <div>
        <Switch>
          {/* Rutas p√∫blicas */}
          <RutaPublica
            exact path='/login'
            component={VistaLogin}
            isAutenticado={usuario.logged}
          />

          {/* Rutas privadas */}
          <RutaPrivada
            path='/'
            component={PanelRutas}
            isAutenticado={usuario.logged}
          />
        </Switch>
      </div>
    </Router>
  )
}