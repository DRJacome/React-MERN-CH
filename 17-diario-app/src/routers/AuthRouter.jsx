import React from 'react'
import { Redirect, Switch, Route } from 'react-router-dom'
import { PantallaLogin } from '../components/auth/PantallaLogin'
import { PantallaRegistro } from '../components/auth/PantallaRegistro'



export const AuthRouter = () => {
  return (
    <div className='auth__main'>
      <div className='auth__box-container'>

      <Switch>
        <Route path='/auth/login' component={PantallaLogin} />
        <Route path='/auth/register' component={PantallaRegistro} />
        <Redirect to='/auth/login' />
      </Switch>
      </div>
    </div>
  )
}