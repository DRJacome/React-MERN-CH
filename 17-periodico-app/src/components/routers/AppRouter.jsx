import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";
import { AuthRouter } from './AuthRouter';
import { PantallaDiario } from '../diario/PantallaDiario';



export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route path='/auth' component={AuthRouter} />
          <Route exact path='/' component={PantallaDiario} />
          <Redirect to='/auth/login' />
        </Switch>
      </div>
    </Router>
  )
}


