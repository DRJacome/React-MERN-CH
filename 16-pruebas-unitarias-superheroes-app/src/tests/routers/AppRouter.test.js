import { mount } from "enzyme"
import { AppRouter } from "../../routers/AppRouter";
import { AutenticacionContext } from "../../auth/AutenticacionContext";

describe('Pruebas en <AppRouter/>', () => {

  const valorContext = {
    dispatch: jest.fn(),
    usuario: {
      logged: false
    }
  }

  test('Debe mostrar el login si no está autenticado', () => {

    const wrapper = mount(
      <AutenticacionContext.Provider value={valorContext}>
        <AppRouter />
      </AutenticacionContext.Provider>);

    console.log(wrapper.html());
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe mostrar el componente Marvel si está autenticado', () => {
    const valorContext = {
      dispatch: jest.fn(),
      usuario: {
        logged: true,
        nombre: 'David'
      }
    }

    const wrapper = mount(
      <AutenticacionContext.Provider value={valorContext}>
        <AppRouter />
      </AutenticacionContext.Provider>);

    console.log(wrapper.html());
    expect(wrapper.find('.navbar').exists()).toBe(true);
  })
})