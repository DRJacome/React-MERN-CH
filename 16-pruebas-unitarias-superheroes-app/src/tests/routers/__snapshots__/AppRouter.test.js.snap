// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> Debe mostrar el login si no está autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      ¡Hola! identifícate
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Identificarse
    </button>
  </div>
</div>
`;
